### PR TITLE
[Quantization][AutoRound] Add NVFP4 linear support

### PR DESCRIPTION
**Pull request by wimCCC**
Supports deployment of AutoRound-format NVFP4 W4A4 quantized models with dense linear layers on CUDA. This PR intentionally covers linear layers only; NVFP4 MoE support will be submitted separately.

### NVFP4 dense on CUDA

`vllm ({'pretrained': 'Qwen3-8B-AutoRound-NVFP4', 'tensor_parallel_size': 1}), gen_kwargs: ({}), limit: None, num_fewshot: 0, batch_size: auto`

|Tasks|Version|Filter|n-shot|Metric|||Value|Stderr|
|-----|------:|------|-----:|------|---|---:|----:|-----:|
|piqa|1|none|0|acc|↑|0.7655|±|0.0099|
|||none|0|acc_norm|↑|0.7715|±|0.0097|

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8688|±  |0.0093|
|     |       |strict-match    |     5|exact_match|↑  |0.8643|±  |0.0094|

### CODE REVIEW

File: tests/quantization/test_auto_round.py [hunk 1]
```diff
@@ -487,6 +498,95 @@ def test_inc_config_accepts_mxfp_family_llm_compressor() -> None:
     assert isinstance(resolve_scheme(layer_config), INCMxfp4Scheme)
 
 
+def test_inc_config_accepts_nvfp4_family_llm_compressor() -> None:
```
Comment (yiliu30) [inline tests/quantization/test_auto_round.py:501]: Please extend the model test scope as well.

File: vllm/model_executor/layers/quantization/inc/schemes/inc_nvfp4_linear.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        layer.register_parameter("input_global_scale", input_global_scale)
+
+        expose_input_quant_key(layer, self.kernel)
+
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        if (
+            torch.unique(layer.input_global_scale).numel() != 1
+            or torch.unique(layer.weight_global_scale).numel() != 1
+        ):
+            logger.warning_once(
```
Comment (yiliu30) [inline vllm/model_executor/layers/quantization/inc/schemes/inc_nvfp4_linear.py:112]: In this case, which scale will be used for the fused layer?

File: vllm/model_executor/layers/quantization/inc/inc.py [hunk 3]
```diff
@@ -54,6 +54,10 @@ class INCConfig(QuantizationConfig):
     MXFP8_DATA_TYPE = "mx_fp"
     MXFP8_PACKING_FORMAT = "auto_round:llm_compressor"
     MXFP8_SUPPORTED_ACT_DTYPES = {"mx_fp", "mx_fp_rceil"}
+    NVFP4_BITS = 4
+    NVFP4_GROUP_SIZE = 16
+    NVFP4_ACT_DATA_TYPE = "nv_fp4_with_static_gs"
+    NVFP4_PACKING_FORMAT = "auto_round:llm_compressor"
```
Comment (yiliu30) [inline vllm/model_executor/layers/quantization/inc/inc.py:60]: The same one as `MXFP8_PACKING_FORMAT` ?

**Comment by yiliu30** [comment]
It would be better to attach the results of gsm8k, thanks!
